docs/rfc/0696-p2p-node.adoc

```diff
@@ -0,0 +1,318 @@
+ip6 = bstr .size 16
+
+; Canonical representation of a peer.
+peer-id: bstr
+
+; Network port.
+port: uint
+----
+
+All types representing requests and responses and their serialisation logic MUST
+be exposed as linkable libraries. It is RECOMMENDED to also expose the
+functionality to communicate with the node via IPC as a library.
+
+== Operations
+
+=== Supervision
+
+Process supervision SHOULD be deferred to established system level service
+managers i.e. `<<systemd>>` and `<<launchd>>` for Linux and macOS respectively.
+To support both long-running as well as ad-hoc usage the daemon implementation
+SHALL be equipped with the ability to detect and read the information from its
+environment necessary to determine if it has been activated via socket. When
+binding to a socket it SHALL use the file descriptors provided by the init
+process. If none are provided it SHALL assume long-running operation and SHALL
+bind to the well-known path on a UNIX domain socket in mode `SOCK_STREAM` under
+Linux:
+
+  $XDG_RUNTIME_DIR/radicle/<srv>-<peer-id>.sock
+
+macOS:
+
+  TODO(xla): Define well-known path for darwin systems.
+
+If sufficiently focused library implementations are present in the ecosystem to
+account for socket activation on the targeted platforms, this document
+RECOMMENDS using/contributing to those options.
+
+Both service managers offer support to fullfil the legacy `inetd` interface.
+Which is deemed insufficient for concenrs over security, lack of support for
+UNIX domain sockets and the design focusing on a process per connection.
+
+==== systemd
+
+Socket activation under systemd is passed on via: 
+
+* `LISTEN_PID` - MUST be equal to the PID of the daemon.
+* `LISTEN_FDS` - Number of received file descriptors, starting at 3 .
+* `LISTEN_NAMES` - Contains colon-separated list of names corresponding to the
+  `FileDescriptorName` option in the service file.
+
+==== launchd
+
+* `LAUNCH_DAEMON_SOCKET_NAME` - Name of the socket according to the `.plist`
+  configuration file.
+
+The name passed to the process MUST be used to check-in with launchd as
+documented in `launch(3)` which in essence involves obtaining the FDs via
+`launch_activate_socket` expecting a name.
+
+=== Configuration
+
+Common service configuration files SHALL be provided alongside the source code
+of the node binary. To support the semi-dynamic nature of one process per
+profile, facalities are required to manage service definitions along the
+lifetime of a profile and when updated need to inform the process management
+system. Such facalities need to be considered wherever profiles are modified,
+i.e. CLI.
+
+The binary SHALL expose all knobs necessary to fine-tune the internal configs of
+the core protocol, i.e. `membership`, `protocol`, `storage`. Additionally
+switches and configuration that subroutines require. As this surface is sizable
+even without subroutines parameters, there SHALL be persistent file based
+configuration in a sane format.
```

Comment:
I presume the "sane format" will be TOML based on yesterday's conversation?

Reply:
Settling on any specific language is not needed since 41b6454.